### DISCUSSION TITLE
cipher: message authentication failed

### DISCUSSION THREAD

**Discussion by israaaa270** [ory/hydra#2315] [Q&A] [answered]
when I run 
-hydra serve all 
I get that error. 
time=2021-01-20T11:49:03+02:00 level=fatal msg=Could not ensure that signing keys for "hydra.openid.id-token" exists. If you are running against a persistent SQL database this is most
likely because your "secrets.system" ("SECRETS_SYSTEM" environment variable) is not set or changed. When running with an SQL database backend you need to make sure that the secret is s
et and stays the same, unless when doing key rotation. This may also happen when you forget to run "hydra migrate sql".. audience=application error=map[message:cipher: message authenti
cation failed] service_name=ORY Hydra service_version=master

I before run  hydra migrate sql .

**Comment by vinckr** [comment, marked as answer]
You ran `hydra migrate sql` before?

`this is most likely because your "secrets.system" ("SECRETS_SYSTEM" environment variable) is not set or changed.`

Did you check your SECRETS?

Are your running Hydra in docker, kubernetes or plain install?

**Reply by israaaa270** [reply]
yes, I ran hydra migrate sql before   -- hydra serve 

yes , I am running  Hydra in docker .
what way  can  I check SECRETS? @vinckr   
Did you mean
`set SYSTEM_SECRET=some-very-random-secret-$§123`

yeah I search about "SECRETS_SYSTEM" ,
`set SECRETS_SYSTEM=this-is-the-primary-secret,this-is-an-old-secret,this-is-another-old-secret`

but still the issue.

**Reply by vinckr** [reply]
Did you export the env variable correctly like [here](https://www.ory.sh/hydra/docs/configure-deploy/#deploy-ory-hydra) for example?:
`export SECRETS_SYSTEM=$(export LC_CTYPE=C; cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)`

Maybe this is also an docker issue with your environmental variables:
[Docker: Set environment variables (-e, --env, --env-file)](https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file)
[Docker Compose: env var](https://docs.docker.com/compose/environment-variables/)

**Reply by israaaa270** [reply]
I run hydra from source code on windows so how this command in windows ,
when I ran it on windows 

`set SECRETS_SYSTEM=$(export LC_CTYPE=C; cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)`

'tr' is not recognized as an internal or external command,
operable program or batch file.

**Reply by vinckr** [reply]
Ah yea those commands are from UNIX. You could run everything in the Windows Linux Subsystem, but that is probably not trivial to set up. 

You can just set environmental variables in your docker run command: 
`docker run --env VAR1=value1`

Could you provide the exact docker commands you are running, any config and logfiles if your problem still persists?

**Reply by israaaa270** [reply]
thanks @vinckr I follow the ory/hydra documentation, I solved the Issues 
and  run hydra serve  successfully on docker command .

But I want to run it from golang terminal that I compile the binary  hydra code  not on docker commands.

**Reply by vinckr** [reply]
Ok, and why is that not working? You set the env variables correctly? (maybe [this](https://superuser.com/questions/949560/how-do-i-set-system-environment-variables-in-windows-10) can help).  
I am not familiar how to run Hydra in Win10 without Docker. It is definitely possible, but you might need to dig a bit deeper on some Windows quirks 
I would really recommend using Docker, or if that is an option for you, an UNIX-based OS.
You could also use the Windows Linux Subsystem, I used it before and it was no problem running Hydra.
 (https://docs.microsoft.com/en-us/windows/wsl/install-win10)

**Reply by israaaa270** [reply]
I run binary hydra from terminal , 
- set variable and get it success as pic:
![image](https://user-images.githubusercontent.com/77671051/105354247-f5df4800-5bf8-11eb-8d95-c1f096a0472a.png)

- `hydra migrate sql --yes $DSN`
- `hydra serve all `      as it.
it stuck at this point :

`level=fatal msg=Could not ensure that signing keys for "hydra.openid.id-token" exists. If you are running against a persistent SQL database this is most likely because your "secrets.system" ("SECRETS_SYSTEM" environment variable) is not set or changed. When running with an SQL database backend you need to make sure that the secret is set and stays the same, unless when doing key rotation. This may also happen when you forget to run "hydra migrate sql".. audience=application error=map[message:cipher: message authentication failed] service_name=ORY Hydra service_version=master`

**Reply by aeneasr** [reply]
Reset/delete the DB and try from the beginning again.

**Reply by israaaa270** [reply]
I tried to run,
- `hydra serve -c /etc/config/hydra/hydra.yml all --dangerous-force-http`      

from terminal in windows: I got this ,

`time=2021-01-24T10:20:48+02:00 level=fatal msg=Unable to instantiate configuration. audience=application error=map[message:GetFileAttributes: The system cannot find the path specified.] service_name=ORY Hydra service_version=master`

**Reply by aeneasr** [reply]
It can not find the file `/etc/config/hydra/hydra.yml`.

**Reply by israaaa270** [reply]
how can I find this file?  Do any command generate this file that I run it before hydra serve ?

**Reply by aeneasr** [reply]
It is the Ory Hydra configuration file. Please check out the [Quickstart](https://www.ory.sh/hydra/docs/5min-tutorial/) (@vinckr could you add a link to or embed the configuration file itself in the docs here?). Maybe it makes sense to do an FAQ item around this as well, and link in the ORY Hydra CLI to that FAQ entry if the config file is not found.

Check out the reference as well: https://www.ory.sh/hydra/docs/reference/configuration

**Reply by israaaa270** [reply]
thank you @aeneasr

**Reply by israaaa270** [reply]
thanks so much @aeneasr  @vinckr

**Reply by israaaa270** [reply]
this error is returned again when run `hydra serve all`:  but returned  `message: malformed ciphertext`

`time=2021-01-31T11:06:03+02:00 level=fatal msg=Could not ensure that signing keys for "hydra.openid.id-token" exists. If you are running against a persistent SQL database this is most likely because your "secrets.system" ("SECRETS_SYSTEM" environment variable) is not set or changed. When running with an SQL database backend you need to make sure that the secret is s
et and stays the same, unless when doing key rotation. This may also happen when you forget to run "hydra migrate sql".. audience=application error=map[message:malformed ciphertext] service_name=ORY Hydra service_version=master`

what are the reason it returned message:malformed ciphertext when it decrypt  in GetKeySet in persister jwk?

**Reply by vinckr** [reply]
Going by the error message, something with the SECRETS_SYSTEM is not working. 
You have the same setup as here: https://github.com/ory/hydra/discussions/2315#discussioncomment-298913 ? 
Did you start with a new SQL database ? 
Did you run `hydra migrate sql` ?

What hydra.yaml are you using now ? 

Can you go through the First Aid again just to make sure it is not a wrong URL or potentially find more information: 
https://www.ory.sh/hydra/docs/debug/#first-aid

**Comment by steowens** [comment]
According to this documentation: https://github.com/ory/hydra

This set of commands ought to get a QuickStart hydra set up:
make docker
docker compose -f quickstart.yml up # ....

Maybe spend a little time on the documentation to clarify the missing pieces that would be known to the development team but unknown to newcomers to hydra who are relying on your documentation?

**Reply by vinckr** [reply]
Hello Steven, 
thanks, yea that is the correct set of commands. 

Did you run into any issues running Ory Hydra?
 
As for documentation, we are always open to requests and feedback as well as any form of contributions - ideally directly in the documentation repository: https://github.com/ory/docs/issues

Thanks!

**Reply by steowens** [reply]
Yes I have tried everything I can find on the web and the only thing that seems to even begin to get Dry Hydra up and running is for me to create my own Dockerfile, and build from source .  All of the other junk out there simply doesn't work.
